pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -421,10 +421,6 @@ def test_setitem_frame_duplicate_columns(self, using_array_manager, request):
             # TODO(ArrayManager) .loc still overwrites
             expected["B"] = expected["B"].astype("int64")
 
-            mark = pytest.mark.xfail(
-                reason="Both 'A' columns get set with 3 instead of 0 and 3"
-            )
-            request.node.add_marker(mark)
```

Comment:
Is this fixed by the code changes?

Reply:
yes

Reply:
So I looked into this to understand _why_ it was actually fixing this. And it's actually because the `copy=True` that gets honored is now masking another bug. 
Inside `dict_to_mgr`, if there are columns not present in the data, we create all-NA arrays for them with `arrays.loc[missing] = [val] * missing.sum()`.

But so that is putting an *identical* array for each columns. And so if you then mutate one columns, all columns get set, _if_ the columns are not copied. With the default of `copy=True` we will now copy the arrays before creating the DataFrame. But with `copy=False` you still have the wrong behaviour.

Reply:
Will open a separate issue for this.

Reply:
-> https://github.com/pandas-dev/pandas/issues/45369